kube-derive/src/custom_resource.rs

```diff
@@ -192,6 +198,109 @@ impl FromMeta for SchemaMode {
     }
 }
 
+/// This struct mirrors the fields of `k8s_openapi::CustomResourceSubresourceScale` to support
+/// parsing from the `#[kube]` attribute.
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Scale {
+    pub(crate) label_selector_path: Option<String>,
+    pub(crate) spec_replicas_path: String,
+    pub(crate) status_replicas_path: String,
+}
+
+// This custom FromMeta implementation is needed for two reasons:
+//
+// - To enable backwards-compatibility. Up to version 0.97.0 it was only possible to set scale
+//   subresource values as a JSON string.
+// - To be able to declare the scale sub-resource as a list of typed fields. The from_list impl uses
+//   the derived implementation as inspiration.
+impl FromMeta for Scale {
+    /// This is implemented for backwards-compatibility. It allows that the scale subresource can
+    /// be deserialized from a JSON string.
+    fn from_string(value: &str) -> darling::Result<Self> {
+        serde_json::from_str(value).map_err(darling::Error::custom)
```

Comment:
I see most of the lines for this impl is showing up as untested. Maybe we can get easy coverage by adding the invocation to kube-derive/tests/crd_schema_test.rs ?

Reply:
Sure thing, I can add some tests to test both the old and the new way.

Reply:
Adding a test proved very valuable! Take a look at cccc052d.

Reply:
That looks sensible. A bit more verbose. But happy for you to play around and improve this!